test/logflare/google/resource_manager_test.exs

```diff
@@ -0,0 +1,193 @@
+defmodule Logflare.Google.CloudResourceManagerTest do
+  use Logflare.DataCase, async: false
+  alias Logflare.Google.CloudResourceManager
+  import Tesla.Mock
+
+  setup do
+    google_configs = Map.new(Application.get_env(:logflare, Logflare.Google))
+    expected_members = setup_test_state()
+
+    %{
+      google_configs: google_configs,
+      expected_members: expected_members
+    }
+  end
+
+  describe "set_iam_policy/0" do
+    test "request body sends expected roles for paying users, paying team users and service accounts",
+         %{
+           google_configs: google_configs,
+           expected_members: expected_members
+         } do
+      mock(fn %{url: url, body: body} ->
+        assert url == expected_url(google_configs)
+
+        assert {[%{"members" => members, "role" => role}], service_accounts} =
+                 body
+                 |> Jason.decode!()
+                 |> get_in(["policy", "bindings"])
+                 |> Enum.split_with(fn %{"role" => role} -> role == "roles/bigquery.jobUser" end)
+
+        assert service_accounts == expected_service_accounts(google_configs)
+        assert role == "roles/bigquery.jobUser"
+        assert members == expected_members
+
+        text("")
+      end)
+
+      CloudResourceManager.set_iam_policy(async: false)
+    end
+  end
+
+  defp expected_url(%{project_number: project_number}),
+    do: "https://cloudresourcemanager.googleapis.com/v1/projects/#{project_number}:setIamPolicy"
+
+  defp expected_service_accounts(google_configs) do
```

Comment:
since this is a sanity check, we don't need to assert on all the members (because we'd be essentially testing implementation in this case).

The main point of the test is to check that external iam policy is called.

it would suffice to check that service accounts length is `>1`, and maybe just assert on one or two special accounts are present.

The length of the test file is almost exceeding the length of actual implementation, which is a maintenance concern

Reply:
this tests that we're setting the correct service account roles hence this assertion because it is actually the thing we're changing with this added editor role 😅